src/inline.h

```diff
@@ -51,6 +51,16 @@ static inline void * append_output_node (struct context * context, size_t size)
   return node -> data;
 }
 
+static inline void * resize_output_node (struct context * context, void * data, size_t size) {
```

Comment:
The alternative I saw to "allocate one large-as-possible node and then shrink it after QOI compression is done" was "allocate lots of little nodes for every QOI chunk", and this seemed better.